grails-app/services/io/xh/toolbox/app/GitHubService.groovy

```diff
@@ -0,0 +1,189 @@
+package io.xh.toolbox.app
+
+import io.xh.hoist.BaseService
+import io.xh.hoist.config.ConfigService
+import io.xh.hoist.http.JSONClient
+import io.xh.hoist.json.JSONSerializer
+import io.xh.toolbox.github.Commit
+import io.xh.toolbox.github.CommitHistory
+import org.apache.http.client.methods.HttpPost
+import org.apache.http.entity.StringEntity
+
+import static io.xh.hoist.util.DateTimeUtils.MINUTES
+
+class GitHubService extends BaseService {
+
+    ConfigService configService
+    Map<String, CommitHistory> commitsByRepo = new HashMap()
+
+    void init() {
+        createTimer(
+            runFn: this.&loadCommitsForAllRepos,
+            interval: 'gitHubCommitsRefreshMins',
+            intervalUnits: MINUTES
+        )
+    }
+
+    /** Return the cached history of commits for a single repo, by name. */
+    CommitHistory getCommitsForRepo(String repoName) {commitsByRepo[repoName]}
+
+    /**
+     * Reload commit history for all configured repos from GitHub API.
+     * @param forceFullLoad - true to drop any already loaded history, false (default) to do an
+     *      incremental load of new commits only.
+     */
+    Map<String, CommitHistory> loadCommitsForAllRepos(Boolean forceFullLoad = false) {
+        def repos = configService.getList('gitHubRepos')
+        withInfo("Refreshing GitHub commits for ${repos.size()} configured repositories") {
```

Comment:
seems a bit chatty to me to have on info

Reply:
It might be - I am still hoping to play with a webhook to have GH ping the server when it has new commits, so we could throttle the refresh way back.  I downgraded all the other logs I had on the service to debug and we can adjust if we come back and find it dominating the logs for no good reason (or up GHService to warn)